core/arch/arm/plat-ls/main.c

```diff
@@ -58,6 +58,9 @@ static struct ns16550_data console_data;
 register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
 			CORE_MMU_PGDIR_SIZE);
 register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE, CORE_MMU_PGDIR_SIZE);
+#if defined(PLATFORM_FLAVOR_lx2160ardb)
+register_ddr(CFG_DRAM0_BASE, CFG_DRAM0_SIZE - CFG_TEE_OS_DRAM0_SIZE);
```

Comment:
Can this be done for all flavors?

Reply:
The variables seem to be common across flavors, but unfortunately I only have an lx2160 around to check.
If someone can confirm the rest of the platforms, we can get rid of the ifdef.

Reply:
we are currently validating the use of dynamic share memory for the LS Soc in our downstream 3.10 code. We will upstream it once validation is done for upstream code as well.
thanks

Reply:
@sdininno, does this mean that you'd rather wait with this PR and that you instead will take care of it all once you're done?

Reply:
@jenswi-linaro ,
no we can go with this PR, we will generalize the dynamic shared memory with another one.

Reply:
I did some testing around and I think we should drop this and re-work the memory model registration for this platform somehow.

The lx2160a has a SO-DIMM, so the amount of memory is not static.
The current OP-TEE port only uses and defines 2GB of memory. So the xtest suite (and the kernel module for that matter), will use the static shared memory and so will any current TAs that require memory sharing. 
What this patch does is allow those 2GB of memory to be shared memory and U-Boot, which only works with dynamic shared memory, starts working. 

But this will break the kernel. The kernel driver(s) will start using dynamic shared memory as well, since it's now registered. If the platform has more than 2GB of memory (which very possible)  the full memory will be scanned and available for the kernel.
In that case no one can guarantee that what every physical memory the kernel will allocate and try to register with OP-TEE lies within the range of that 2GB region we registered. So the xtest and any application requiring memory sharing will fail.

From what I can tell it's TF-A that does the scanning. So if we want to use U-Boot we need to define a way of dynamically registering allowed ddr ranges for OP-TEE once the scan is finished.